src/components/Login/LoginContainer.tsx

```diff
@@ -99,32 +99,34 @@ export const InitialisationContainer: FunctionComponent<NavigationProps> = ({
     setLoginStage("SCAN");
   };
 
-  const shouldResendAlert = async (): Promise<boolean> => {
+  const getResendConfirmationIfNeeded = async (): Promise<boolean> => {
     return new Promise(resolve => {
-      Alert.alert(
-        "Resend OTP?",
-        lastResendWarningMessageRef.current,
-        [
-          {
-            text: "CANCEL",
-            onPress: () => resolve(false)
-          },
-          {
-            text: "RESEND",
-            onPress: () => resolve(true)
-          }
-        ],
-        { cancelable: false }
-      );
+      if (!lastResendWarningMessageRef.current) resolve(true);
+      else {
```

Comment:
Minor nit: can we use the braces here since we're having the else clause already.